packages/beeq/src/components/page-title/bq-page-title.tsx

```diff
@@ -0,0 +1,174 @@
+  @State() private hasSubTitle = false;
+
+  // Public Property API
+  // ========================
+
+  // Prop lifecycle events
+  // =======================
+
+  /** If true, the page title back button will be shown */
+  @Prop({ reflect: true }) haveBackNavigation: boolean;
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /** Handler to be called when page title navigation button loses focus */
+  @Event() bqBackBlur: EventEmitter<HTMLBqPageTitleElement>;
+
+  /** Handler to be called when page title navigation button is clicked */
+  @Event() bqBackClick: EventEmitter<HTMLBqPageTitleElement>;
+
+  /** Handler to be called when page title navigation button is focused */
+  @Event() bqBackFocus: EventEmitter<HTMLBqPageTitleElement>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  // Listeners
+  // ==============
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private handleClick = (ev: Event) => {
+    ev.stopPropagation();
+    this.bqBackClick.emit(this.el);
+  };
+
+  private handleBlur = (ev: Event) => {
+    ev.stopPropagation();
+    this.bqBackBlur.emit(this.el);
+  };
+
+  private handleFocus = (ev: Event) => {
+    ev.stopPropagation();
+    this.bqBackFocus.emit(this.el);
+  };
+
+  private handleSlotChange = () => {
+    this.hasSuffix = hasSlotContent(this.suffixElem, 'suffix');
+    this.hasSubTitle = hasSlotContent(this.subTitleElem, 'sub-title');
+  };
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    return (
+      <div class="flex flex-col" part="wrapper">
+        <div class="flex gap-xs [padding-block:--paddingY]" part="base">
+          {/* Back navigation button */}
+          <div class={{ flex: true, '!hidden': !this.haveBackNavigation }} part="back">
+            <slot name="back">
+              <bq-button
+                appearance="link"
+                part="btn-back"
+                exportparts="button"
+                onBqBlur={this.handleBlur}
+                onBqClick={this.handleClick}
+                onBqFocus={this.handleFocus}
+              >
+                <bq-icon
+                  color="text--primary"
+                  name="arrow-left"
+                  size="24"
+                  weight="bold"
+                  part="icon"
+                  exportparts="base,svg"
+                />
+              </bq-button>
+            </slot>
+          </div>
```

Comment:
After taking a better look into accessibility and how screen readers react to this new header, I think we should completely let the Back Navigation button up to the consumer:

``` suggestion
          {/* Back navigation button */}
          <div class={{ flex: true, '!hidden': !this.haveBackNavigation }} part="back">
            <slot name="back" />
          </div>
```

Then in Storybook, we can change a bit the Template:

```diff
  return html`
    <bq-page-title
      ?have-back-navigation=${args['have-back-navigation']}
      @bqBackClick=${args.bqBackClick}
      @bqBackFocus=${args.bqBackFocus}
      @bqBackBlur=${args.bqBackBlur}
    >
+      ${args['have-back-navigation']
+        ? html`
+           <bq-button appearance="link" slot="back">
+             <bq-icon
+                color="text--primary"
+                name="arrow-left"
+                weight="bold"
+                role="img"
+                title="Navigate back to the previous page"
+              ></bq-icon>
+            </bq-button>
+          `
+        : nothing}
      ${args.title} ${args['sub-title'] ? html`<div slot="sub-title">${args['sub-title']}</div>` : nothing}
      ${args.actions ? html`<div class="${actionsSlotClass}" slot="suffix">${args.actions}</div>` : nothing}
      ${customDivider}
    </bq-page-title>
  `;
```

By removing the predefined back button, we no longer need the events and their handlers 🙂 


https://github.com/Endava/BEEQ/assets/328492/5672e257-b7e2-484c-a5ea-1f48f371894d

Reply:
You could also think about removing the `haveNavigationBack` property and handle the slot container display based in the slot content, same as you did in the sub-title and divider slot containers 😉